benchmark/nixlbench/contrib/Dockerfile

```
@@ -200,10 +204,17 @@ RUN rm -rf $VIRTUAL_ENV && uv venv $VIRTUAL_ENV --python $DEFAULT_PYTHON_VERSION
 ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 # Install python dependencies
 RUN uv pip install --upgrade meson meson-python pybind11 patchelf pyYAML click tabulate auditwheel tomlkit
-# Install PyTorch
-RUN CUDA_SHORT_VERSION=cu$(echo $CUDA_VERSION | cut -d. -f1,2 | tr -d .) && \
-    FLAGS="--index-url https://download.pytorch.org/whl/$CUDA_SHORT_VERSION" && \
-    uv pip install $FLAGS torch torchvision torchaudio
+# Install PyTorch: use system torch if present, else install into venv.
+# When system has torch, add its site-packages via a .pth file so the venv sees it with no workarounds.
+RUN if /usr/bin/python${DEFAULT_PYTHON_VERSION} -c "import torch" 2>/dev/null; then \
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)